### PR TITLE
feat: Add workflows client methods. Fixes #1

### LINKED ISSUE #1: Add workflows client methods
Add go methods for the workflows API endpoints found in the Pipekit docs: https://docs.pipekit.io/pipekit-api

### PR DESCRIPTION
**Pull request by JPZ13**
This PR:
- Adds utility methods for making http requests
- Adds pipe (workflow) client methods

The PR is a little larger than I would have liked

### CODE REVIEW

**Review by isubasinghe — APPROVED** [review]

**Comment by isubasinghe** [comment]
I couldn't really find anything wrong, good work!

File: backend.go [hunk 1]
```diff
@@ -143,3 +182,15 @@ func formatJSONHeader(ctx context.Context, req *http.Request) {
...
+// FormatURLPath is a convenience method for ensuring that paths are properly
+// formatted
+func FormatURLPath(format string, params ...string) string {
+	// Convert parameters to interface{} and URL-escape them
+	untypedParams := make([]interface{}, len(params))
+	for i, param := range params {
+		untypedParams[i] = interface{}(url.QueryEscape(param))
+	}
+
+	return fmt.Sprintf(format, untypedParams...)
```
Comment (isubasinghe) [inline backend.go:195]: Is the interface{} here because there maybe other types later on? 
Why cannot it be just []string instead of []interface{}?

Comment (JPZ13) [reply backend.go:195]: Good call. Will change to string

Comment (JPZ13) [reply backend.go:195]: Never mind - Sprintf takes an interface type and kicks a type error for an array of strings

Comment (isubasinghe) [reply backend.go:195]: Oh right sorry, I am on 1.18beta and that is probs why it worked for me. something to do with generics probably

**Pull request merged**